### PR TITLE
8277481: Obsolete seldom used CDS flags

### PR DESCRIPTION
**Pull request by hseigel**
Please review this change to obsolete deprecated CDS options UseSharedSpaces, RequireSharedSpaces, DynamicDumpSharedSpaces, and DumpSharedSpaces.  The change was tested by running Mach5 tiers 1-2 on Linux, Mac OS, and Windows and Mach5 tiers 3-5 on Linux x64 and Windows x64.

The use of UseSharedSpaces in ps_core_common.c was tested on Mac OS x64 by temporarily removing serviceability/sa/ClhsdbPmap.java#core from the problem list.

Thanks! Harold

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277481](https://bugs.openjdk.java.net/browse/JDK-8277481): Obsolete seldom used CDS flags

### Reviewers
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to 3f6c6deefbc2184ae9ae7397b1f0097defc55d11
 * [Calvin Cheung](https://openjdk.java.net/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to 3f6c6deefbc2184ae9ae7397b1f0097defc55d11
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6800/head:pull/6800` \
`$ git checkout pull/6800`

Update a local copy of the PR: \
`$ git checkout pull/6800` \
`$ git pull https://git.openjdk.java.net/jdk pull/6800/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6800`

View PR using the GUI difftool: \
`$ git pr show -t 6800`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6800.diff">https://git.openjdk.java.net/jdk/pull/6800.diff</a>

</details>

### CODE REVIEW

**Review by iklam — APPROVED** [review]
Looks good to me. Thanks for fixing this!

File: src/jdk.hotspot.agent/share/native/libsaproc/ps_core_common.c [hunk 1]
```diff
@@ -300,7 +300,7 @@ bool init_classsharing_workaround(struct ps_prochandle* ph) {
       jvm_name = lib->name;
       useSharedSpacesAddr = lookup_symbol(ph, jvm_name, USE_SHARED_SPACES_SYM);
       if (useSharedSpacesAddr == 0) {
-        print_debug("can't lookup 'UseSharedSpaces' flag\n");
+        print_debug("can't lookup 'UseSharedSpaces' symbol\n");
```
Comment (calvinccheung) [inline src/jdk.hotspot.agent/share/native/libsaproc/ps_core_common.c:303]: Maybe the `print_debug` at line 311 should also be updated from "flag" to "symbol"?

**Review by calvinccheung — APPROVED** [review]
Looks good. Just one nit.

File: src/jdk.hotspot.agent/share/native/libsaproc/ps_core_common.c (hunk 1 above)
Comment (hseigel) [reply src/jdk.hotspot.agent/share/native/libsaproc/ps_core_common.c:303]: fixed.  Thanks for pointing it out.

**Review by dholmes-ora — APPROVED** [review]
LGTM!

Thanks,
David

**Comment by hseigel** [comment]
Thanks Ioi, Calvin, and David for the reviews!

/integrate

**Pull request closed without merging**